### ISSUE TITLE
Feature request - Max amount of DKP

### ISSUE THREAD

**Issue by perrosenlind** [Roeshambo/MonolithDKP#303]
Hi, 

Idk if this is the place for a request, but I would love to have an option that sets maximum allowed DKP. Is this possible?

**Comment by lantisnt** [comment]
With dkp decay set there always is some softcap, eg with 20% decay and max +30dkp/week its 120

**Comment by perrosenlind** [comment]
> With dkp decay set there always is some softcap, eg with 20% decay and max +30dkp/week its 120

So if I want to set the max dkp to 100, with a hard cap, how can I do that?

**Comment by lantisnt** [comment]
Generally the math behind this is really basic:
`DKP(n + 1) = DECAY*DKP(n) + MAX_DKP_PER_WEEK`

Where:
* DKP(0) = 0 (no initial DKP, but for the limit it does not matter)
* DECAY in <0,1> where 1 = 100%, 0 = 0%, 0.8 = 20% (1 - 0.8 = 0.2 = 20% decay)
* MAX_DKP_PER_WEEK is the value of total dkp one can get per week without spending anything

Here is example of our guilds math, with MAX_DKP = 30, DECAY = 0.8 (20%):
https://www.wolframalpha.com/input/?i=RSolve%5B%7Bd%5Bn%5D+%3D%3D+%280.8*d%5Bn+-+1%5D+%2B+30%29%2C+d%5B0%5D+%3D%3D+0%7D%2C+d%5Bn%5D%2C+n%5D

From it you receive a non-recursive equation like ours `-150 5^(-1 n) (1 2^(2 n) - 1 5^n)` (Result row, just open it in new window to get it into wolfram):
https://www.wolframalpha.com/input/?i=-150+5%5E%28-1+n%29+%281+2%5E%282+n%29+-+1+5%5En%29&assumption=%22ClashPrefs%22+-%3E+%7B%22Math%22%7D

now you need to get the math limit from it, so wrap it around with:
`Limit[<the equation>, n -> infinity]`, click enter and you'll get:  https://www.wolframalpha.com/input/?i=Limit%5B-150+5%5E%28-1+n%29+%281+2%5E%282+n%29+-+1+5%5En%29%2C+n+-%3E+infinity%5D

From this you will get value **before decay** (150 for us) so manually apply your decay, and you will have your total limit. For us `0.8*150 = 120` soft cap

So basically what you need to do is to try tweaking those values in a way you will get the 100 at the end (inverse math of it is super hard to explain thus its easier to do it manually).

You can also calculate it in excel: (copy it to yourself)
https://docs.google.com/spreadsheets/d/12vet00pEuApubkgZnJYLXOqUHX6pQqQIl1rc3eOsMvg/edit?usp=sharing

From my basic tests it looks like 25DKP /week and 20% decay result in soft cap of 100 DKP

**Comment by perrosenlind** [comment]
So from what I understand of this, there's no function to just handle a hard max-dkp cap? I would prefer to not use any type of decay at this point to handle this.
